lib/markdown_videos/renderer.rb

```diff
@@ -13,42 +13,26 @@ def initialize(string, options = {})
     end
 
     def render
-      MarkdownVideos::SERVICES.keys.each do |service|
-        @string = render_service(service)
-      end
-      @string
-    end
+      @string.gsub(/!\[([^\]]*)\]\(([^)]+)\)/) do
+        rendered = nil
+        match_data = Regexp.last_match
```

Comment:
Personally, I think that giving an argument to the gsub block is better than using Regexp.last_match.